builder/processor/src/main/java/io/helidon/builder/processor/GenerateAbstractBuilder.java

```diff
@@ -0,0 +1,1294 @@
+        pw.println(";");
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("@Override");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("public int hashCode() {");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        if (equalityFields.isEmpty()) {
+            // no fields on this type
+            if (hasSuper) {
+                pw.println("return super.hashCode();");
+            } else {
+                // hashcode is a constant, as there are no fields and no super type
+                pw.println("return 1;");
+            }
+        } else {
+            if (hasSuper) {
+                pw.print("return 31 * super.hashCode() + Objects.hash(");
+            } else {
+                pw.print("return Objects.hash(");
+            }
+
+            pw.print(equalityFields.stream()
+                             .map(PrototypeProperty::name)
+                             .collect(Collectors.joining(", ")));
+            pw.println(");");
+        }
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+    }
+
+    private static void toString(PrintWriter pw,
+                                 TypeContext typeContext,
+                                 String typeName,
+                                 boolean hasSuper,
+                                 List<CustomMethods.CustomMethod> prototypeMethods,
+                                 boolean isBuilder) {
+        if (prototypeMethods.stream()
+                .map(CustomMethods.CustomMethod::generatedMethod)
+                .map(CustomMethods.GeneratedMethod::method)
+                .filter(it -> "toString".equals(it.name()))
+                .filter(it -> it.returnType().equals(STRING_TYPE))
+                .anyMatch(it -> it.arguments().isEmpty())) {
+            // do not create toString() if defined as a custom method
+            return;
+        }
+        // only create to string if not part of prototype methods
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("@Override");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("public String toString() {");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print("return \"");
+        pw.print(typeName);
+
+        List<PrototypeProperty> toStringFields = typeContext.propertyData()
+                .properties()
+                .stream()
+                .filter(PrototypeProperty::toStringValue)
+                .toList();
+
+        String prefix = SOURCE_SPACING.repeat(6);
+        if (toStringFields.isEmpty()) {
+            pw.println("{}\"");
+        } else {
+            pw.println("{\"");
+        /*
```

Comment:
remove

Reply:
removed the comment, that is probably what you asked for (it was not valid anymore)